tests/std/tests/P0798R8_monadic_operations_for_std_optional/test.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <optional>
+#include <type_traits>
+#include <utility>
+
+using namespace std;
+
+struct Immovable {
+    constexpr Immovable(int x) : value(x) {}
+    Immovable(const Immovable&) = delete;
+    Immovable(Immovable&&)      = delete;
+    Immovable& operator=(const Immovable&) = delete;
+    Immovable& operator=(Immovable&&) = delete;
+    constexpr ~Immovable() {}
+
+    int value;
+};
+
+template <class T>
+struct fn {

```

Comment:
I observe that this test covers only function objects, not pointers-to-members, so it isn't verifying that `and_then()` and `transform()` are using `invoke()`.

It's not *strictly* necessary since I can see `invoke()` in the implementation and there's little risk of this regressing, but I think that adding basic coverage that each overload accepts a pointer-to-member-function or pointer-to-member-data could be a good idea. (No need to test both PMFs and PMDs - just one will indicate that `invoke()` is being used.)